File: Java/HammingDistance.java
/**
 * Created by Ethan on 2017/9/17.
 */
public class HammingDistance {
    public static int hammingDistance(int x, int y) {
        int dis = 0;
        int xor = x ^ y;
        for (; xor > 0; xor >>= 1){
            if ((xor & 1) == 1) {
                dis++;
            }
        }
        return dis;
    }

    public static void main(String[] args) {
        System.out.println(hammingDistance(1, 4));   // 2
        System.out.println(hammingDistance(12, 25)); // 3
    }
}

File: SwiftCode/HammingDistance.swift
//
//  HammingDistance.swift
//  SwiftCode
//
//  Created by Luminoid Li on 1/6/23.
//

import Foundation

//@main
private class Solution {
    // Time complexity: O(1)
    // Space complexity: O(1)
    
    func hammingDistance(_ x: Int, _ y: Int) -> Int {
        return (x ^ y).nonzeroBitCount
    }
    
    static func main() {
        let solution = Solution()
        print(solution.hammingDistance(1, 4))   // 2
        print(solution.hammingDistance(3, 1))   // 1
        print(solution.hammingDistance(12, 10)) // 2
    }
}
